bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -105,49 +119,78 @@ pub mod pallet {
 		/// Convert versioned location failure
 		UnsupportedLocationVersion,
 		/// Check location failure, should start from the dispatch origin as owner
-		OwnerCheck,
+		InvalidAssetOwner,
 		/// Send xcm message failure
 		Send,
 		/// Withdraw fee asset failure
 		FundsUnavailable,
+		/// Convert to reanchored location failure
+		LocationConversionFailed,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Call `create_agent` to instantiate a new agent contract representing `origin`.
-		/// - `origin`: Can be any sovereign `Location`
+		/// - `origin`: `Location` can be from either of:
+		/// 	a. any kind of location from a sibling parachain
+		/// 	b. sudo from local, location will be `Here` in this case.
```

Comment:
See my comment below, I think there is a high chance your assumption that `governance == Here` is incorrect.